packages/yak-swc/yak_swc/src/lib.rs

```diff
@@ -630,10 +696,19 @@ where
     for decl in &mut n.decls {
       if let Pat::Ident(BindingIdent { id, .. }) = &decl.name {
         let previous_variable_name = self.current_variable_name.clone();
+        
+        // Check if this variable will be default-exported
+        let will_be_default_exported = self.variables.is_default_exported(&id.to_id());
+        
         self.current_variable_name = Some(ScopedVariableReference::new(
           id.to_id(),
           vec![id.sym.clone()],
         ));
+        
+        if will_be_default_exported {
+          self.current_exported = true;
```

Comment:
```suggestion
          self.current_exported = self.variables.is_default_exported(&id.to_id());
```

And the if + the temporary variable aren't used anymore